ee/allowedcmd/run_disclaimed_darwin.go

```diff
@@ -0,0 +1,144 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+int responsibility_spawnattrs_setdisclaim(posix_spawnattr_t attrs, int disclaim);
+
+int spawn_disclaimed(const char *path, char *const argv[], char *const envp[]) {
+    posix_spawnattr_t attrs;
+    int err = posix_spawnattr_init(&attrs);
+    if (err != 0) {
+        return err;
+    }
+
+    // Set the disclaim attribute
+    err = responsibility_spawnattrs_setdisclaim(&attrs, 1);
+    if (err != 0) {
+        posix_spawnattr_destroy(&attrs);
+        return err;
+    }
+
+    pid_t pid;
+    // passing NULL for file_actions, we capture stderr/out from running this as subcommand
+    err = posix_spawn(&pid, path, NULL, &attrs, argv, envp);
+    posix_spawnattr_destroy(&attrs);
+    if (err != 0) {
+        return err;
+    }
+
+    int status;
+    if (waitpid(pid, &status, 0) == -1) {
+        return -1;
+    }
+
+    // WIFEXITED will return non-zero if the child process terminated normally
+    return WIFEXITED(status) ? WEXITSTATUS(status) : -1;
+}
+*/
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/user"
+	"unsafe"
+
+	"github.com/kolide/launcher/pkg/log/multislogger"
+)
+
+func RunDisclaimed(_ *multislogger.MultiSlogger, args []string) error {
+	ctx := context.Background()
+	cmd, err := commandToDisclaim(ctx, args)
+	if err != nil {
+		return fmt.Errorf("gathering subcommand: %w", err)
+	}
+
+	cmdPath := C.CString(cmd.Path)
+	defer C.free(unsafe.Pointer(cmdPath))
+
+	// we're building up the C arguments here like []*C.char{cmdPath, arg1, arg2..., nil}
+	// note that cmdPath will already be present as the first argument
+	cargs := make([]*C.char, len(cmd.Args)+1) // 1 extra for null terminator
+	for i, arg := range cmd.Args {
+		cargs[i] = C.CString(arg)
+	}
+	cargs[len(cargs)-1] = nil // should be terminated by null entry
+
+	defer func() {
+		for _, arg := range cargs {
+			if arg != nil {
+				C.free(unsafe.Pointer(arg))
+			}
+		}
+	}()
+
+	cenvs := make([]*C.char, len(cmd.Environ())+1) // 1 extra for null terminator
+	for i, arg := range cmd.Environ() {
+		cenvs[i] = C.CString(arg)
+	}
+	cenvs[len(cenvs)-1] = nil // should be terminated by null entry
+
+	defer func() {
+		for _, arg := range cenvs {
+			if arg != nil {
+				C.free(unsafe.Pointer(arg))
+			}
+		}
+	}()
+
+	ret := C.spawn_disclaimed(cmdPath, &cargs[0], &cenvs[0])
+	if int(ret) != 0 {
+		return fmt.Errorf("spawn_disclaimed returned error code %d", int(ret))
+	}
+
+	return nil
+}
+
+func commandToDisclaim(ctx context.Context, args []string) (*TracedCmd, error) {
+	if len(args) != 1 {
+		return nil, fmt.Errorf("rundisclaimed expects 1 subcommand, received args of length %d", len(args))
+	}
+
+	subcommand := args[0]
+	switch subcommand {
+	case "brew":
+		return generateBrewCommand(ctx)
+	default:
+		return nil, errors.New("unsupported command for rundisclaimed")
+	}
+}
+
+func generateBrewCommand(ctx context.Context) (*TracedCmd, error) {
+	cmd, err := Brew(ctx, "outdated", "--json")
```

Comment:
We don't care about not being able to be able to control these brew args via command-line args to rundisclaimed?

Reply:
not for brew currently (this was the only usage i saw for that allowedcmd) but I think that's a good callout in general (others will have varied usage)- we can just whitelist the command itself as the first arg and then allow arbitrary args through. since this is running disclaimed I don't see any additional risk there if that makes sense to you?

Reply:
I think we're going to want the be able to specify these.